src/Nethermind/Nethermind.BeaconNode/AttestationProducer.cs

```diff
@@ -0,0 +1,126 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Nethermind.Core2;
+using Nethermind.Core2.Containers;
+using Nethermind.Core2.Crypto;
+using Nethermind.Core2.Types;
+
+namespace Nethermind.BeaconNode
+{
+    public class AttestationProducer
+    {
+        private readonly BeaconChainUtility _beaconChainUtility;
+        private readonly BeaconStateAccessor _beaconStateAccessor;
+        private readonly BeaconStateTransition _beaconStateTransition;
+        private readonly ICryptographyService _cryptographyService;
+        private readonly IForkChoice _forkChoice;
+        private readonly ILogger _logger;
+        private readonly IStore _store;
+
+        public AttestationProducer(ILogger<AttestationProducer> logger,
+            ICryptographyService cryptographyService,
+            BeaconChainUtility beaconChainUtility,
```

Comment:
I made it an interface recently - shall we always use interfaces in constructors for any services / utilities? helps with testing

Reply:
yes, please :D

Reply:
May not be merged yet, as I don't see any interfaces (did you mean IBeaconChainUtility ?)

I agree that interfaces are better, so can fix after merge.

Reply:
No change to make.